In [1]:
getwd()

[1] "/home/jovyan/work/swaggest-pino"

In [2]:
list.files()

[1] "Figuring it out.ipynb"       "Individual Planning Project"
[3] "loadingstuff.ipynb"          "players.csv"                
[5] "README.md"                   "sessions.csv"               
[7] "Untitled.ipynb"

In [3]:
library(tidyverse)
sessions_original <- read_csv("sessions.csv")
players_original <- read_csv("players.csv")

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter:

In [4]:
print(players_original, rows = 10)

# A tibble: 196 × 7
   experience subscribe hashedEmail              played_hours name  gender   Age
   <chr>      <lgl>     <chr>                           <dbl> <chr> <chr>  <dbl>
 1 Pro        TRUE      f6daba428a5e19a3d475748…         30.3 Morg… Male       9
 2 Veteran    TRUE      f3c813577c458ba0dfef809…          3.8 Chri… Male      17
 3 Veteran    FALSE     b674dd7ee0d24096d1c0196…          0   Blake Male      17
 4 Amateur    TRUE      23fe711e0e3b77f1da7aa22…          0.7 Flora Female    21
 5 Regular    TRUE      7dc01f10bf20671ecfccdac…          0.1 Kylie Male      21
 6 Amateur    TRUE      f58aad5996a435f16b0284a…          0   Adri… Female    17
 7 Regular    TRUE      8e594b8953193b26f498db9…          0   Luna  Female    19
 8 Amateur    FALSE     1d2371d8a35c8831034b25b…          0   Emer… Male      21
 9 Amateur    TRUE      8b71f4d66a38389b7528bb3…          0.1 Nata… Male      47
10 Veteran    TRUE      bbe2d83de678f519c4b3daa…          0   Nyla  Female    22
# ℹ 186 

In [5]:
players_original <- tibble(players_original)

In [6]:
player_data <- players_original |>
    select(experience, subscribe, played_hours, gender, Age)
player_data

experience subscribe played_hours gender            Age
1   Pro         TRUE     30.3         Male               9 
2   Veteran     TRUE      3.8         Male              17 
3   Veteran    FALSE      0.0         Male              17 
4   Amateur     TRUE      0.7         Female            21 
5   Regular     TRUE      0.1         Male              21 
6   Amateur     TRUE      0.0         Female            17 
7   Regular     TRUE      0.0         Female            19 
8   Amateur    FALSE      0.0         Male              21 
9   Amateur     TRUE      0.1         Male              47 
10  Veteran     TRUE      0.0         Female            22 
11  Veteran     TRUE      1.6         Female            23 
12  Pro         TRUE      0.0         Male              17 
13  Beginner    TRUE      1.5         Female            25 
14  Amateur     TRUE      0.2         Male              22 
15  Veteran     TRUE      0.0         Non-binary        17 
16  Amateur    FALSE      0.0         Male              22 
17  Beginner    TRUE      0.0         Female            17 
18  Amateur     TRUE     48.4         Female            17 
19  Amateur     TRUE      0.5         Male              17 
20  Regular     TRUE      0.6         Male              19 
21  Regular     TRUE      0.3         Male              58 
22  Amateur    FALSE      0.1         Female            17 
23  Beginner    TRUE      1.0         Male              17 
24  Beginner    TRUE      0.0         Male              17 
25  Amateur     TRUE      0.7         Female            21 
26  Regular     TRUE      0.6         Female            28 
27  Regular     TRUE      0.0         Male              17 
28  Veteran    FALSE      0.0         Male              23 
29  Amateur     TRUE      1.8         Male              23 
30  Veteran    FALSE      0.1         Male              18 
⋮   ⋮          ⋮         ⋮            ⋮                 ⋮  
167 Amateur    FALSE      1.0         Male              26 
168 Beginner   FALSE      0.3         Female            17 
169 Veteran     TRUE      0.1         Male              17 
170 Veteran    FALSE      0.1         Male              17 
171 Pro        FALSE      0.4         Male              21 
172 Beginner   FALSE      1.8         Male              32 
173 Veteran     TRUE      0.0         Agender           20 
174 Regular    FALSE      0.0         Male              50 
175 Veteran     TRUE      0.0         Male              17 
176 Amateur     TRUE      0.3         Male              17 
177 Pro        FALSE      0.0         Non-binary        17 
178 Veteran    FALSE      2.7         Non-binary        21 
179 Amateur     TRUE      0.4         Two-Spirited      17 
180 Veteran     TRUE      0.1         Non-binary        17 
181 Amateur    FALSE      2.1         Male              24 
182 Amateur     TRUE      0.8         Female            22 
183 Pro        FALSE      0.2         Male              17 
184 Amateur     TRUE     32.0         Male              22 
185 Pro        FALSE      1.7         Male              17 
186 Regular    FALSE      0.1         Male              18 
187 Veteran     TRUE      0.1         Female            44 
188 Amateur     TRUE      0.0         Male              17 
189 Beginner    TRUE      0.0         Female            17 
190 Amateur    FALSE      0.0         Male              17 
191 Amateur     TRUE      0.0         Male              20 
192 Amateur     TRUE      0.0         Female            17 
193 Veteran    FALSE      0.3         Male              22 
194 Amateur    FALSE      0.0         Prefer not to say 57 
195 Amateur    FALSE      2.3         Male              17 
196 Pro         TRUE      0.2         Other             NA

In [7]:
set.seed(42)
library(rsample)
library(tidymodels)

player_split <- initial_split(player_data, prop = 0.70)
player_training <- training(player_split) |>
    mutate(subscribe = as.factor(subscribe))
player_testing <- testing(player_split) |>
    mutate(subscribe = as.factor(subscribe))

player_training

── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──

✔ broom        1.0.6     ✔ recipes      1.1.0
✔ dials        1.3.0     ✔ tune         1.1.2
✔ infer        1.0.7     ✔ workflows    1.1.4
✔ modeldata    1.4.0     ✔ workflowsets 1.0.1
✔ parsnip      1.2.1     ✔ yardstick    1.3.1

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filter()   masks stats::filter()
✖ recipes::fixed()  masks stringr::fixed()
✖ dplyr::lag()      masks stats::lag()
✖ yardstick::spec() masks readr::spec()
✖ recipes::step()   masks stats::step()
• Use suppressPackageStartupMessages() to eliminate package startup messages



experience subscribe played_hours gender            Age
1   Veteran    TRUE      12.5         Agender           27 
2   Amateur    TRUE       0.1         Male              23 
3   Beginner   TRUE       0.2         Prefer not to say 17 
4   Veteran    TRUE       0.0         Male              22 
5   Veteran    FALSE      0.1         Male              19 
6   Beginner   TRUE       0.1         Male              24 
7   Pro        TRUE       0.2         Other             NA 
8   Amateur    TRUE       0.0         Male              21 
9   Veteran    TRUE       0.1         Male              17 
10  Beginner   TRUE       0.0         Male              17 
11  Regular    TRUE       0.2         Male              14 
12  Pro        TRUE       0.5         Male              12 
13  Beginner   TRUE       0.0         Female            17 
14  Regular    TRUE       0.0         Male              22 
15  Amateur    TRUE       0.1         Male              17 
16  Regular    TRUE       0.6         Male              19 
17  Beginner   TRUE       0.1         Male              17 
18  Beginner   FALSE      0.0         Female            17 
19  Amateur    TRUE       0.0         Male              22 
20  Amateur    TRUE      56.1         Male              23 
21  Regular    TRUE       5.6         Male              17 
22  Amateur    TRUE      32.0         Male              22 
23  Regular    TRUE       0.0         Male              17 
24  Regular    TRUE       0.5         Female            20 
25  Veteran    FALSE      0.0         Male              20 
26  Regular    TRUE       0.1         Male              21 
27  Veteran    FALSE      0.0         Non-binary        17 
28  Beginner   TRUE       0.0         Non-binary        17 
29  Beginner   TRUE       2.0         Prefer not to say 27 
30  Veteran    FALSE      0.0         Male              17 
⋮   ⋮          ⋮         ⋮            ⋮                 ⋮  
108 Regular    TRUE        0.0        Male              17 
109 Veteran    TRUE        0.0        Male              17 
110 Amateur    FALSE       0.0        Male              17 
111 Pro        TRUE        1.2        Male              17 
112 Pro        FALSE       0.0        Non-binary        17 
113 Amateur    TRUE        0.1        Male              19 
114 Veteran    FALSE       2.7        Non-binary        21 
115 Amateur    FALSE       0.0        Two-Spirited      26 
116 Beginner   TRUE        0.0        Male              22 
117 Amateur    TRUE        0.1        Male              17 
118 Amateur    TRUE        0.7        Female            21 
119 Pro        TRUE        1.0        Male              17 
120 Veteran    FALSE       0.0        Male              46 
121 Amateur    TRUE        0.2        Male              22 
122 Veteran    TRUE        0.0        Male              18 
123 Amateur    TRUE        0.0        Male              17 
124 Beginner   FALSE       0.0        Male              28 
125 Regular    TRUE        1.5        Male              20 
126 Veteran    FALSE       1.4        Prefer not to say 25 
127 Amateur    TRUE        0.1        Male              23 
128 Amateur    FALSE       0.0        Prefer not to say 33 
129 Regular    TRUE      223.1        Male              17 
130 Beginner   TRUE        0.0        Female            17 
131 Amateur    TRUE        1.2        Non-binary        17 
132 Veteran    TRUE        0.0        Non-binary        17 
133 Pro        TRUE        0.1        Male              15 
134 Beginner   TRUE        0.0        Male              17 
135 Beginner   TRUE        1.0        Male              17 
136 Regular    FALSE       0.0        Male              18 
137 Pro        TRUE        0.0        Male              17

**1) NUMBER OF OBSERVATIONS AND VARIABLES IN players.csv DATASET**

In [22]:
#number of observations/variables (given from source, unaltered):
n_players.csv <- nrow(players_original)
p_players.csv <- ncol(players_original)
cat("players.csv:", n_players.csv, "rows,", p_players.csv, "columns\n")

#number of observations/variables (what I will be using for my analysis):
n_players <- nrow(player_data)
p_players <- ncol(player_data)
cat("player data to be used:", n_players, "rows,", p_players, "columns\n")
print("(excluded email and name; irrelevant variables to the question)")

players.csv: 196 rows, 7 columns
player data to be used: 196 rows, 5 columns
[1] "(excluded email and name; irrelevant variables to the question)"


Therefore, in players.csv, there are: 
- 196 observations
- 7 variables

**2) Variable names, types, and descriptions**

In [30]:
desc_of_vars <- c(
    "How the player identifies in gaming experience level",
    "Whether player has subscribed to the newsletter (True or false)",
    "Email address of player",
    "Total time spent in the game (in hours)",
    "Name of player",
    "Gender of player",
    "Age of player (in years)")
   
description_players <- tibble(
    variable = names(players_original),
    type = map_chr(players_original, ~class(.x)[1]),
    description = desc_of_vars)

description_players

variable     type     
experience   experience   character
subscribe    subscribe    logical  
hashedEmail  hashedEmail  character
played_hours played_hours numeric  
name         name         character
gender       gender       character
Age          Age          numeric  
             description                                                    
experience   How the player identifies in gaming experience level           
subscribe    Whether player has subscribed to the newsletter (True or false)
hashedEmail  Email address of player                                        
played_hours Total time spent in the game (in hours)                        
name         Name of player                                                 
gender       Gender of player                                               
Age          Age of player (in years)